extensions/http/jetty/src/main/java/org/eclipse/dataspaceconnector/extension/jetty/JettyService.java

```diff
@@ -148,16 +152,26 @@ private ServerConnector httpsServerConnector(int port) {
         var sslConnectionFactory = new SslConnectionFactory(contextFactory, "http/1.1");
         var sslConnector = new ServerConnector(server, httpConnectionFactory(), sslConnectionFactory);
         sslConnector.setPort(port);
+        configure(sslConnector);
         return sslConnector;
     }
 
     @NotNull
     private ServerConnector httpServerConnector(int port) {
         ServerConnector connector = new ServerConnector(server, httpConnectionFactory());
         connector.setPort(port);
+        configure(connector);
         return connector;
     }
 
+    private void configure(ServerConnector connector) {
+        connectorConfigurationCallbacks.stream().forEachOrdered(c -> c.accept(connector));
```

Comment:
since `connectorConfigurationCallback` is already a list and is therefore ordered, I think `forEach` would work too, right?

Reply:
Formally not, see JavaDoc: "the behavior of this operation is explicitly nondeterministic." It probably works in practice, but we should use `forEachOrdered` to have the explicit guarantee.

Reply:
okay then, `forEachOrdered` it is :)

Reply:
@algattik I think Paul was referring to using `connectorConfigurationCallbacks.forEach()` which is equivalent to a "for" loop iterating the list in order. No need to create a stream here.